lib/bloc/user_profile/user_profile_bloc.dart

```diff
@@ -1,33 +1,74 @@
 import 'dart:async';
 import 'dart:io';
 
+import 'package:breez_translations/breez_translations_locales.dart';
+import 'package:c_breez/bloc/user_profile/default_profile_generator.dart';
 import 'package:c_breez/bloc/user_profile/user_profile_state.dart';
 import 'package:c_breez/models/user_profile.dart';
 import 'package:c_breez/services/breez_server.dart';
 import 'package:c_breez/services/notifications.dart';
+import 'package:fimber/fimber.dart';
 import 'package:hydrated_bloc/hydrated_bloc.dart';
 import 'package:path_provider/path_provider.dart';
 
 const PROFILE_DATA_FOLDER_PATH = "profile";
 
+final _log = FimberLog("UserProfileBloc");
+
 class UserProfileBloc extends Cubit<UserProfileState> with HydratedMixin {
   final BreezServer _breezServer;
   final Notifications _notifications;
 
-  UserProfileBloc(this._breezServer, this._notifications)
-      : super(UserProfileState.initial());
+  UserProfileBloc(
+    this._breezServer,
+    this._notifications,
+  ) : super(UserProfileState.initial()) {
+    stream.listen((profile) {
+      _log.v("State: ${profile.profileSettings.toJson()}");
+      final settings = profile.profileSettings;
+      if (settings.color == null || settings.animal == null) {
+        _log.v("Profile has not color or name, generating new random ones…");
+        final defaultProfile = generateDefaultProfile();
+        emit(profile.copyWith(
+          profileSettings: settings.copyWith(
+            color: settings.color ?? defaultProfile.color,
+            animal: settings.animal ?? defaultProfile.animal,
+          ),
+        ));
+      } else if (settings.name == null) {
+        _log.v("Profile has no name, generating new one…");
+        emit(profile.copyWith(
+          profileSettings: settings.copyWith(
+            name: DefaultProfile(
+              settings.color!,
+              settings.animal!,
+            ).buildName(getSystemLocale()),
+          ),
+        ));
+      } else if (!settings.registered) {
+        _log.v("Profile has not been registered, registering…");
+        registerForNotifications();
```

Comment:
@ademar111190  Also don't we need to register for push notifications on every restart? The push token may change over time.

Reply:
Good point, you are rigth, I'll change the logic here.

Reply:
Fixed